pkg/database/database.go

```diff
@@ -86,49 +84,32 @@ func (db *Database) Ping(ctx context.Context) error {
 	return db.db.PingContext(ctx)
 }
 
-func (db *Database) QueryResources(ctx context.Context, kind, apiVersion, limit, continueId, continueDate string) ([]*unstructured.Unstructured, int64, string, error) {
-	var pq *paramQuery
+func (db *Database) QueryResources(ctx context.Context, kind, apiVersion, namespace, name, limit, continueId, continueDate string) ([]*unstructured.Unstructured, int64, string, error) {
+	// To simplify the logic, we don't use `else` and instead override `pq` when needed
+	pq := db.newParamQuery(db.info.GetResourcesLimitedSQL())
+	pq.addStringParams(kind, apiVersion, limit)
 
-	if continueId != "" && continueDate != "" {
-		pq = db.newParamQuery(db.info.GetResourcesLimitedContinueSQL())
-		pq.addStringParams(kind, apiVersion, continueDate, continueId, limit)
-	} else {
-		pq = db.newParamQuery(db.info.GetResourcesLimitedSQL())
-		pq.addStringParams(kind, apiVersion, limit)
-	}
-
-	return db.performResourceQuery(ctx, pq)
-}
-
-func (db *Database) QueryNamespacedResources(ctx context.Context, kind, apiVersion, namespace, limit, continueId, continueDate string) ([]*unstructured.Unstructured, int64, string, error) {
-	var pq *paramQuery
-
-	if continueId != "" && continueDate != "" {
-		pq = db.newParamQuery(db.info.GetNamespacedResourcesLimitedContinueSQL())
-		pq.addStringParams(kind, apiVersion, namespace, continueDate, continueId, limit)
-	} else {
+	if namespace != "" {
 		pq = db.newParamQuery(db.info.GetNamespacedResourcesLimitedSQL())
 		pq.addStringParams(kind, apiVersion, namespace, limit)
+
+		if name != "" {
```

Comment:
Can you only query using a `name` if you've used a `namespace` too?

Reply:
Yes, to retrieve by name you need to specify namespace as we don't support cluster wide resources.

And yes, `limit` is not compatible with `name`. Retrieving by name always retrieve one element or it fails if there is more than one element.